server/http/handlers/service.go

```diff
@@ -301,3 +309,9 @@ func (s *service) newDocumentFromBytes(p jsonparser.Parser, data []byte) ([]byte
 	}
 	return namespace, doc, nil
 }
+
+const (
+	batchSizeBucketVersion = 1
+	bucketSize             = 1000
```

Comment:
Might make sense to use a `bucketSize` of 200 and `numBucket` of 20 to start with.